docs/Modules/common-modules/event-bus.md

```diff
@@ -9,4 +9,174 @@ parent: "Common Modules"
+
+  // prevent event default
+  event.preventDefault();
+
+  // Some custom logic
+  // [...]
+});
+```
+
+
+## once
+
+### Usage
+
+once(events: _String_ or _Array< string >_, [priority=1000]: _number_, callback: _Function_, [that]:_Object_): _undefined_
+
+### Parameters
+
+* `events`: name of the event or _Array_ of event names.
+* `priority`: optional parameter specifying the priority with which the listener is called, where a higher number represents a higher priority and therefore leads to the listener being called earlier. If not provided, default of `1000` will be used.
+* `callback`: function which will be invoked with `event, ...additionalArguments`. Returning anything but `undefined` from this will stop the listener propagation. Returning `false` will prevent the execution of the default action for that event.
+*  `that`: optional parameter which can be used to pass context (`this`) to the callback.
+
+### Return Value
+
+_undefined_
+
+### Code Example
+
+```javascript
+// listener performing some one-time operation
+function listener(event) { return 'One-time operation finished'; }
+
+eventBus.once('onceEvent', listener);
+
+const firstEventFired = eventBus.fire('onceEvent'),
+      secondEventFired = eventBus.fire('onceEvent');
+
+console.log(firstEventFired); // 'One-time operation finished'
+console.log(secondEventFired); // undefined
+```
+
+## off
+
+### Usage
+
+off(events: _String_ or _Array< string >_, [callback]: _Function_): _undefined_
+
+### Parameters
+
+* `events`: name of the event or _Array_ of event names.
+* `callback`: optional parameter specifying the callback listener to be removed.
+
+### Return Value
+
+_undefined_
+
+### Code Example
+
+```javascript
+function fooListener(event) {
+  console.log('Foo');
+}
+
+function barListener(event) {
+  console.log('Bar');
+}
+
+eventBus.on('foo', fooListener);
+eventBus.on('foo', barListener);
+
+eventBus.fire('foo'); // 'Foo' 'Bar'
+
+eventBus.off('foo', fooListener);
+
+eventBus.fire('foo'); // 'Bar'
+
+eventBus.off('foo');
+
+eventBus.fire('foo'); // ''
+```
+
+## createEvent
+
+### Usage
+
+createEvent(data: _Object_): _Object_
+
+### Parameters
+
+* `data`: the event data.
+
+### Return Value
+
+An _InternalEvent_ object which can be passed to [fire](#fire).
+
+## fire
+
+### Usage
+
+fire(name: _String_, [data]: _Object_, [arguments]: _...Object_): _Object_
+
+### Parameters
+
+* `name`: event name.
+* `data`: optional _InternalEvent_ object. If this is not already an instance of `InternalEvent`, the `fire` method will internal wrap it as such (using [`createEvent`](createEvent)).
+* `arguments`: optional additional arguments to be passed to the callback functions.
+
+### Return Value
+
+* `false` if the default action was prevented by listeners.
+* `undefined` if no listeners were registered for that event.
+* _Object_ if the invoked listener with the highest priority returns a value.
+
+### Code Example
+
+```javascript
+function listener(...args) {
```

Comment:
Maybe we should promote the way we use the event bus internally in this example. If the first parameter is an object it will be assigned to the event itself which allows you to do

```javascript
eventBus.on('foo', ({ answer }) => console.log(answer));

eventBus.fire('foo', { answer: 42 });
```

At least this is the way we do it everywhere in our toolkit.

Reply:
Good point. I would still like to keep the existing example, since then both examples represent basically two alternatives of how to use `fire`